docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+# Context Menu Manager (API)
+
+### Table of content
+- [Introduction](#Introduction)
+    - [Context menu UI behavior](#Context menu UI behavior)
+    - [Context-Menu config structure](#Context Menu config structure)
+    - [Structure tree](#Structure tree)
+- [API methods]()
+  - [Create a group](#Create a group)
+  - [Update a group](#Update a group)
+  - [Delete a group](#Delete a group)
+  - [Add item](#Add item)
+  - [Update item](#Update item)
+  - [Delete item](#Delete item)
+  - [Get config](#Get config)
+- [Understanding the usage flow]()
+  - [Registration of new items](#Registration of new items)
+  - [Triggering the context menu flow](#Triggering the context menu flow)
+- [Example of usage]()
+
+## Introduction
+The context menu manager provides a convenient way for managing the context menu structure.
+The manager is responsible for building a tree of nested groups/items that forms the actual structure of the context menu, as presented in the UI.   
+The manager is based on CRUD principles while providing simple functions for adding, updating, deleting, and in the end - provides a function for reading the main tree that have been created.
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
+
+* Groups themselves are not visually displayed, they just displays their items and separate between them.
+
+* A group can display a title, if a title is specified.
```

Comment:
```suggestion
* A group can have a title.
```
Not sure how important it is

Reply:
Again - the importance here is for understanding how to use it and why it was built (the API) like this.
A group is not visually displayed, not like an item that t is displayed always. But a group **can** display a title, only if the title is specified. otherwise - the group won't take any place at all. When one comes to using this API, he must understand this.

Reply:
As the developer of the API, you don't know how people will consume it and/or show the UI for it